Author: Simon Frankenberger <simon-alpine@fraho.eu>
Subject: Disable failing tests
Upstream: Yes (https://bugs.openjdk.org/browse/JDK-8299245)

"Some gtests do not work well in non-debug build"
Disable failing tests as workaround

--- old/test/hotspot/gtest/runtime/test_os_linux.cpp
+++ new/test/hotspot/gtest/runtime/test_os_linux.cpp
@@ -439,6 +439,7 @@
 
 #if !defined(__clang_major__) || (__clang_major__ >= 5) // DWARF does not support Clang versions older than 5.0.
 // Test valid address of method ReportJNIFatalError in jniCheck.hpp. We should get "jniCheck.hpp" in the buffer and a valid line number.
+#if 0
 TEST_VM(os_linux, decoder_get_source_info_valid) {
   char buf[128] = "";
   int line = -1;
@@ -447,6 +448,7 @@
   ASSERT_TRUE(strcmp(buf, "jniCheck.hpp") == 0);
   ASSERT_TRUE(line > 0);
 }
+#endif
 
 // Test invalid addresses. Should not cause harm and output buffer and line must contain "" and -1, respectively.
 TEST_VM(os_linux, decoder_get_source_info_invalid) {
@@ -466,6 +468,7 @@
 
 // Test with valid address but a too small buffer to store the entire filename. Should find generic <OVERFLOW> message
 // and a valid line number.
+#if 0
 TEST_VM(os_linux, decoder_get_source_info_valid_overflow) {
   char buf[11] = "";
   int line = -1;
@@ -474,9 +477,11 @@
   ASSERT_TRUE(strcmp(buf, "<OVERFLOW>") == 0);
   ASSERT_TRUE(line > 0);
 }
+#endif
 
 // Test with valid address but a too small buffer that can neither store the entire filename nor the generic <OVERFLOW>
 // message. We should find "L" as filename and a valid line number.
+#if 0
 TEST_VM(os_linux, decoder_get_source_info_valid_overflow_minimal) {
   char buf[2] = "";
   int line = -1;
@@ -485,6 +490,7 @@
   ASSERT_TRUE(strcmp(buf, "L") == 0); // Overflow message does not fit, so we fall back to "L:line_number"
   ASSERT_TRUE(line > 0); // Line should correctly be found and returned
 }
+#endif
 #endif // clang
 
 #ifdef __GLIBC__
